packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -193,6 +229,17 @@ export function createInstance(
   if (__DEV__) {
     // TODO: take namespace into account when validating.
     const hostContextDev = ((hostContext: any): HostContextDev);
+    if (__DEV__ && enableEventAPI) {
+      if (hostContextDev.isEventTarget) {
+        (hostContextDev: any).hostNodeCount++;
```

Comment:
Hm. This won't work. Mutation in here is not safe because we could do multiple passes over the parent or render them out of order in a Suspense case.

It might be better to place this warning in the place where you actually attach the event listeners and the error happens. I'm not sure where you're planning to attach the listeners but if you do it in the commit phase and plan on traversing the Fibers to find the DOM node, then you can do the warning there too. Another possible place is in the complete phase for the EventTarget, because that's when all children are guaranteed to have been completed.